### DISCUSSION TITLE
[MultiApps] Setting second sub-application's variable ICs by 'initial_from_file_var' or 'SolutionUserObject' from 1st sub-application's latest results

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#24324] [Q&A General] [answered]
Dear all,

I use 'type = TimePeriod' setting of [Controls] blocks to allow different sub-applications in [MultiApps] blocks to run in time sequence. 

I want to set the second sub-application's variable ICs by 'initial_from_file_var' or 'SolutionUserObject' from 1st sub-application's latest results. But all the sub-applications set their variables' initialization at the beginning of the whole simulation and just get the initial results from corresponding exodus files. 

I tried the 'global_time_offset' and 'wait_for_first_app_init=true' but I could not get my expected results.

Could you please help me with this? 

Thanks a lot.
J

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can just chain the subapps using the execute_on flags. 
An app executing on timestep_end of another app will execute after. This lets you stagger the solves and transfer from one app to another before. 

Global_time_offset can be used as well to execute a some time steps in one app before starting the other, but you have to turn on catch_up as well iirc to force the two applications to synchronize in time. 

Guillaume
